clang/lib/3C/StructInit.cpp

```diff
@@ -16,10 +16,15 @@
 
 using namespace clang;
 
-bool StructVariableInitializer::variableNeedsInitializer(VarDecl *VD) {
-  if (VD->getStorageClass() == StorageClass::SC_Extern)
+bool StructVariableInitializer::variableNeedsInitializer(DeclaratorDecl *DD) {
+  assert("Struct variable initializer should only look at records!" &&
```

Comment:
Does this assert need to be here? Can you just return false?

I'd also prefer to keep the types as specific as possible. It looks like they were changed to simplify the conditional on line 44 below? If you incorporate `isStructOrUnionType` into `variableNeedsInitializer` you can add an `isa<VarDecl>` and avoid the re-typing. Would that work?

Reply:
I've moved things around a bit so that it's no longer an assert. With regards to the type, this is as specific as possible. `FieldDecl` doesn't subclass `VarDecl`, so for this to work recursively on struct members, it needs to take a `DeclaratorDecl`.